apko-publish/action.yaml

```
@@ -138,17 +144,18 @@ outputs:
       The digest of the published container image.
 
 runs:
-  using: docker
-  image: "docker://ghcr.io/wolfi-dev/apko:latest"
-  env:
-    # Set up go-containerregistry's GitHub keychain.
-    GITHUB_ACTOR: ${{ inputs.repository_owner }}
-    GITHUB_TOKEN: ${{ inputs.token }}
-    REPOSITORY: ${{ inputs.repository }}
-  entrypoint: /bin/sh
-  args:
-    - '-c'
-    - |
+  using: composite
+  steps:
+  - shell: bash
+    run: |
+      docker run -i --rm --entrypoint /bin/sh \
+        -v $PWD:/workspace \
+        --workdir /workspace \
+        -e "GITHUB_ACTOR=${{ inputs.repository_owner }}" \
+        -e "GITHUB_TOKEN=${{ inputs.token }}" \
+        -e "REPOSITORY=${{ inputs.repository }}" \
```

I'm unsure how the Github env might be used here.. if you want to be safe you could pass in the whole env into an envfile:
```suggestion
    run: |
      trap "rm -f github.env" EXIT
      env > github.env
      echo "GITHUB_ACTOR=${{ inputs.repository_owner }}" >> github.env
      echo "GITHUB_TOKEN=${{ inputs.token }}" >> github.env
      echo "REPOSITORY=${{ inputs.repository }}" >> github.env
      docker run -i --rm -env-file github.env --entrypoint /bin/sh \
        -v $PWD:/workspace \
        --workdir /workspace \
```

---

I don't love the idea of passing the full environment through, and it comes with wrinkles like `PATH` 😅 

I added these explicitly because these were the only values set from the `env` block, so the only things missing would be additional environment variables Github is implicitly setting for docker steps today (we needed none for the build version fwiw)

---

YOLO!